contracts/cw-proposal-multiple/src/proposal.rs

```diff
@@ -0,0 +1,177 @@
+    pub fn update_status(&mut self, block: &BlockInfo) {
+        self.status = self.current_status(block);
+    }
+
+    /// Returns true iff this proposal is sure to pass (even before
+    /// expiration if no future sequence of possible votes can cause
+    /// it to fail). Passing in the case of multiple choice proposals
+    /// means that one of the options that is not "None of the above" has won the most votes.
+    pub fn is_passed(&self) -> bool {
+        // Proposal can only pass if quorum has been met.
+        if does_vote_count_pass(
+            self.votes.total(),
+            self.total_power,
+            self.voting_strategy.get_quorum(),
+        ) {
+            let (choice_idx, winning_choice) = self.calculate_winning_choice();
+            // Check that the winning choice is not None.
+            if winning_choice.option_type != MultipleChoiceOptionType::None {
+                // If the leading choice cannot possibly be outwon by the "None" option, the proposal has passed.
+                // This means that the difference between them must be greater than the remaining vote power.
+                let none_count = u128::from(self.get_none_vote_count());
+                let winning_choice_count = self.votes.vote_weights[choice_idx].u128();
+                let remaining_power = self.total_power.sub(self.votes.total()).u128();
+                if winning_choice_count - none_count > remaining_power {
+                    return true;
+                }
+            }
+        }
+
+        false
+    }
+
+    pub fn is_rejected(&self, block: &BlockInfo) -> bool {
+        match (
+            does_vote_count_pass(
+                self.votes.total(),
+                self.total_power,
+                self.voting_strategy.get_quorum(),
+            ),
+            self.expiration.is_expired(block),
+        ) {
+            // Quorum is met and proposal is expired.
+            (true, true) => {
+                // Proposal is rejected if "None" is the winning option.
+                let (_, winning_choice) = self.calculate_winning_choice();
+                if winning_choice.option_type == MultipleChoiceOptionType::None {
+                    return true;
+                }
+
+                // Proposal is rejected if all of the options are tied.
+                if let Some(first) = self.votes.vote_weights.first() {
+                    return self.votes.vote_weights.iter().all(|x| *x == *first);
+                }
+                false
+            }
+            // Quorum is met and proposal has not expired OR Quorum is not met and proposal is not expired.
+            (true, false) | (false, false) => {
+                // Proposal is rejected if "None" has the majority of the total power because there is no way for
+                // another option to outnumber it.
+                let none_vote_count = Uint128::from(self.get_none_vote_count());
+                does_vote_count_fail(
```

Comment:
Should this be `does_vote_count_pass`? We're checking if `None` has passed the majority threshold right?